runtime/runtime_impl_jetty12/src/main/java/com/google/apphosting/runtime/jetty/JettyServletEngineAdapter.java

```diff
@@ -138,10 +137,12 @@ public void run(Runnable runnable) {
       httpConfiguration.setSendDateHeader(false);
       httpConfiguration.setSendServerVersion(false);
       httpConfiguration.setSendXPoweredBy(false);
+      httpConfiguration.setUriCompliance(UriCompliance.LEGACY);
```

Comment:
Note also that there are two parts to URI legacy behaviour.  This compliance will allow ambiguous URIs into the container. You also need to `ServletHandler.setDecodeAmbiguousURIs(true)` if you wish such URIs to be returned from `getServletPath` and `getPathInfo` as there were previously

Reply:
I have added some logic to set `setDecodeAmbiguousURIs` to true in the EE10 `AppEngineWebAppContext#newServletHandler` if `LEGACY_MODE` is set.